geps/gep-2643/index.md

```diff
@@ -0,0 +1,514 @@
+
+# GEP-2643: TLS/SNI based routing / TLSRoute
+
+* Issue: [#2643](https://github.com/kubernetes-sigs/gateway-api/issues/2643)
+* Status: Standard
+
+## TLDR
+
+This GEP documents the implementation of a route type that uses the Server Name
+Indication attribute (aka SNI) of a TLS handshake to make the routing decision.
+
+While this feature is also known sometimes as TLS passthrough, where after the 
+server name is identified, the gateway does a full encrypted passthrough of the 
+communication, this GEP will also cover cases where a TLS communication is 
+terminated on the Gateway before being passed to a backend.
+
+## Goals
+
+* Provide a TLS route type, based on the SNI identification.
+* Support both Passthrough (core support) and Termination (extended support) TLS modes for TLSRoute
+* Provide load balancing of a TLS route, allowing a user to define a route, 
+based on the SNI identification that should pass the traffic to N load balanced backends
+
+## Longer Term Goals
+* Implement capabilities for [ESNI](https://www.cloudflare.com/learning/ssl/what-is-encrypted-sni/)
+* Implement capabilities for [ECH](https://blog.cloudflare.com/encrypted-client-hello/)
+* Extend TLSRouteRule to support hostname being a matcher
+
+## Non-Goals
+* Provide an interface for users to define different listeners or ports for the
+`TLSRoute` - This will be covered by the [ListenerSet enhancement](https://gateway-api.sigs.k8s.io/geps/gep-1713/).
+* Describe re-encryption capabilities when TLS Termination is being used - This will
+be covered by [GEP-4274](https://github.com/kubernetes-sigs/gateway-api/issues/4274), extending
```

Comment:
nit: We haven't decided that this will be covered.
```suggestion
* Describe re-encryption capabilities when TLS Termination is being used - This will
be proposed in [GEP-4274](https://github.com/kubernetes-sigs/gateway-api/issues/4274), extending
```